### DISCUSSION TITLE
Renaming Simplenote text files so the SN importer recognizes the note titles

### DISCUSSION THREAD

**Discussion by clark800** [standardnotes/forum#2505] [General] [answered]
Simplenote doesn't seem to export tags currently, but it does export plain text files, so it is fairly straightforward to adapt these for the text file importer. A separate import script would still be helpful to streamline the import and prevent issues with titles that contain characters that are not legal in filenames.

For now, I was able to use this script to rename the text files to be the first line of the document so that the Standard Notes importer would get the correct titles rather than the hashes that Simplenote uses. Usage: first extract the Simplenote zip, then pass the extracted folder path to this script on the command line.

```
#!/bin/sh
  
for filepath in "$1"/*.txt; do
  title=$(head -n 1 "$filepath")
  mv $filepath "$1/$title.txt"
done
```

**Comment by atmoio** [comment]
Thanks for the snippet!

**Comment by clark800** [comment]
No problem. By the way, this leaves the titles inside the notes; if you want to remove them you can use `tail -n +2` and `rm` instead of `mv`.

**Comment by atmoio** [comment]
Updated Simplenote import instructions here: https://standardnotes.org/help/30/how-can-i-import-my-notes-from-simplenote

**Comment by shawnyeager** [comment]
I just downloaded the app, and went through the above-noted instructions to import from Simplenote. I'm still seeing hashes as note titles. I read this thread to mean that the technique @clark800 references was to have been integrated. Incorrect?

**Comment by atmoio** [comment]
Ah sorry, I can see the confusion. I just posted that link as a general update so people searching Simplenote can find it. But no, the snippet above hasn't been integrated. You'd need to do this manually.

**Comment by compassdrl** [comment]
Where do I place the folder path on the script?

**Comment by myreli** [comment, marked as answer]
The SimpleNote `notes.zip/source/notes.json` export doesn't include filenames, unfortunately. This is why SN can't detect them. So there are two options: 

1. If you don't care about filenames, follow [these instructions](https://standardnotes.com/help/30/how-can-i-import-my-notes-from-simplenote), which will import all notes from the `notes.json` as is
2. If _do_ you care about preserving the filenames, SimpleNote also includes the notes as text files, you can use the [plain text importer](https://standardnotes.com/help/11/how-can-i-import-notes-from-a-bunch-of-text-files)

Tested on the most recent web client, `v3.149.5`, both methods work. Let us know if you have any issues!
